### ISSUE TITLE
ListView title localization

### ISSUE THREAD

**Issue by yumitoshka** [SubPointSolutions/spmeta2#928]
I have problems with listview title localization. It seems that SP ignores TitleResource property for listview.

We have multilanguage teamsite with russian default language and english as alternative language.
I'm using the following model to provision the listview:

```cs
public static ListViewDefinition LastTwentyFiveRequests = new ListViewDefinition
        {
            Title = "Запросы", // title in russian language
            TitleResource = new List<ValueForUICulture>
            {
                new ValueForUICulture{CultureId = 1033, Value = "Requests view"}
            },
            Url = "RequestsURL.aspx",
            RowLimit = 25,
            Query = "<OrderBy><FieldRef Name='ID' Ascending='False'/></OrderBy>",
            IsDefault = true,
            Fields = RequestViewFields
        };
```

After provisioning I get the following behaviour:
1. When I use russian language, listview title displays like "RequestsURL", so it used url for title
2. When I use english language, listview title displays like "Запросы" (default title of the view in russian language)

I tried to add all values for different cultures into the TitleResourses property, but it didn't make any changes in behaviour.

For other artifacts (lists, fields, content types, navigation) localization works perfectly.
Am I doing something wrong? Or is it a bug in SPMeta or SP?
May be there are another ways to localize listview title using spmeta? 

P.S.:
SSOM 15.0.4569.1000
SPMeta2 1.2.16278.0757

**Comment by SubPointSupport** [comment]
A bit weird, indeed.

SPMet2 maps .Title to SPView.Title, and then collection from .TitleResource straight to SPView.TitleResource.

Would you please give a try for the following code as well? Adding two values for .TitleResource collection for both languages potentially may solve the issue. Overwise, we'd need to get into root of the issue.

```cs
TitleResource = new List<ValueForUICulture>
{
new ValueForUICulture{CultureId = 1033, Value = "Requests view"},
new ValueForUICulture{CultureId = 1049, Value = "Your Russian title"}
}
```

**Comment by yumitoshka** [comment]
Hi!

I've tried this code, but it didn't make any change. I see "RequestsUrl" in russian language and "RequestsTitle" (value from Title property) in english language. 
But when I create site with english language by default it displays "RequestsTitle" for both languages.

It seems that SharePoint doesn't use TitleResource property.

**Comment by SubPointSupport** [comment]
Getting even weirder.

Leave it with us, will get a closer look over the week. Hard to say why and what's there. Will test this one trying to replicate the case. Meanwhile, just to ensure you've got the language packs installed right? Plus languages are made available for the sites as well.

Alternatively, we may suggest to try this model on different farms / VMs to eliminate the config. Then use CSOM to eliminate the API. Potentially could be a misconfiguration of the SharePoint itself or some bugs, indded.  We are going to do that on our end as well. Just in case you have that ability, that would be great. 

As for the provision implementation, we don't do much with localizations mapping out ValueForUICulture values to the XXXResource property - same way as you were to implement with the native C# code. The implementation for list view is here: 

https://github.com/SubPointSolutions/spmeta2/blob/master/SPMeta2/SPMeta2.SSOM/ModelHandlers/ListViewModelHandler.cs

We've got a feeling that you would have the right localization if you remove the .Url property of the list view. The thing is that SSOM API does not provide a way to create a list view with the URL. If URL is defined, we create a list view with the Title = URL. That forces SharePoint to create a new list view with not only with the Title = URL but also with the list view URL as we like to. Later, the list view gets Title renamed with the "definition.Title" value. You can see that hack here - https://github.com/SubPointSolutions/spmeta2/blob/master/SPMeta2/SPMeta2.SSOM/ModelHandlers/ListViewModelHandler.cs#L157

```cs
currentView = targetList.Views.Add(
                            string.IsNullOrEmpty(listViewModel.Url) ? listViewModel.Title : GetSafeViewUrl(listViewModel.Url),
                            viewFields,
                            listViewModel.Query,
                            (uint)listViewModel.RowLimit,
                            listViewModel.IsPaged,
                            listViewModel.IsDefault,
                            viewType,
                            isPersonalView);

currentView.Title = listViewModel.Title;
```
It seems that in your case SharePoint remembers the first value of the Title (which would be "RequestsURL" from GetSafeViewUrl(listViewModel.Url) call so that the further renamings don't work with localization or something. No matter how many time .Title is changed, somehow only the first one gets rendered as you suggested. That's extremely weird. Need more testing from our side.

**Comment by yumitoshka** [comment]
> Meanwhile, just to ensure you've got the language packs installed right? Plus languages are made available for the sites as well
Yes, you are absolutely right. 

> We've got a feeling that you would have the right localization if you remove the .Url property of the list view.
I've tried this one. As a result I get one title for both locales and TitleResources are ignored as it was before.

Later I will try to use CSOM and will write you about the results. But unfortunetely I don't have ability to try different farms and VMs. 

Anyway, thank you for your investigation and detailed explanation!

**Comment by SubPointSupport** [comment]
@yumitoshka, is this issue still active? We are running more tests around that issue. Would you like us to get into further details?

**Comment by SubPointSupport** [comment]
Closing. Will reopen once needed.

**Comment by sp-ru** [comment]
Hello,
I experience the same issue. Title and title resources property setup like below:
Title = "Имя",
TitleResource = new List<ValueForUICulture> {
    new ValueForUICulture { CultureId = 1033, Value = "Name"},
    new ValueForUICulture { CultureId = 1049, Value = "Имя"}
    },
Url = "ViewUrl.aspx",

Result: view's name in web browser still ViewUrl, but in SPManager 2013 Title property is correct and view's node name renders in russain. 

This issue prevents migration of our solutions to SPMeta2. Please reopen issue.

About environment:
SP2013 SP1 en + ru language pack, SSOM, SPMeta2 1.2.120.

**Comment by SubPointSupport** [comment]
Absolutely, moving to the current split. That's going to be the first thing to figure out, then.
